MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/offline/CacheManagementActivity.java

```diff
@@ -0,0 +1,299 @@
+    setContentView(R.layout.activity_offline_cache_manipulation);
+
+    newMaxAmbientCacheSizeTextView = findViewById(
+        R.id.set_max_ambient_cache_size_button_editText);
+
+    // Disabling Firebase Remote Config usage here to avoid build and compiling issues during
+    // development.
+    if (!BuildConfig.DEBUG) {
+
+      // Get an instance of Firebase Remote Config.
+      firebaseRemoteConfig = FirebaseRemoteConfig.getInstance();
+
+      // Sets the minimum interval between successive fetch calls. Fetches less than
+      // number of seconds passed in as a parameter, after the last fetch from the Firebase Remote
+      // Config server would use values returned during the last fetch.
+      // More info at https://bit.ly/2XJdvuT.
+
+      // NOT recommended for apps in production. Uncomment to use for app development and testing
+      /*firebaseRemoteConfig.setConfigSettingsAsync(new FirebaseRemoteConfigSettings.Builder()
+          .setMinimumFetchIntervalInSeconds(5)
+          .build());*/
+
+      // Set default Remote Config parameter values. The app uses the in-app default values, and
+      // when you need to adjust those defaults, you set an updated value for only the values you
+      // want to change in the Firebase console.
+      firebaseRemoteConfig.setDefaults(R.xml.cache_management_remote_config_defaults);
+
+      // Fetch the latest Remote Config values from Firebase and then adjust the cache if needed.
+      firebaseRemoteConfig.fetchAndActivate().addOnCompleteListener(this,
+          new OnCompleteListener<Boolean>() {
+            @Override
+            public void onComplete(@NonNull Task<Boolean> task) {
+              if (task.isSuccessful()) {
+                adjustCacheBasedOnRemoteConfigServerSettings();
+              } else {
+                Logger.d("CacheManagementActivity", "Couldn\'t get Firebase cache "
+                    + "management remote config settings from Firebase.");
+              }
+            }
+          });
+    }
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(@NonNull MapboxMap mapboxMap) {
+        mapboxMap.setStyle(Style.MAPBOX_STREETS, new Style.OnStyleLoaded() {
+          @Override
+          public void onStyleLoaded(@NonNull Style style) {
+
+            findViewById(R.id.invalidate_ambient_cache_button).setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                invalidateAmbientCache();
+              }
+            });
+
+            findViewById(R.id.set_max_ambient_cache_size_button).setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                long newMaxAmbientCacheSize = Long.valueOf(
```

Comment:
If the edit text is empty, this will throw a `NumberFormatException`.

Reply:
Ok, fixed to

```java
 findViewById(R.id.set_max_ambient_cache_size_button).setOnClickListener(new View.OnClickListener() {
              @Override
              public void onClick(View view) {
                try {
                  long newMaxAmbientCacheSize = Long.valueOf(
                      newMaxAmbientCacheSizeTextView.getText().toString());
                  setNewMaxAmbientCache(newMaxAmbientCacheSize);
                } catch (NumberFormatException exception) {
                  Logger.d("CacheManagementActivity", "%s", exception);
                }
              }
            });
```